core/src/main/java/com/datastax/oss/driver/api/core/metadata/Metadata.java

```diff
@@ -36,11 +37,45 @@
  */
 public interface Metadata {
   /**
-   * The nodes known to the driver, indexed by the address that it uses to connect to them. This
-   * might include nodes that are currently viewed as down, or ignored by the load balancing policy.
+   * The nodes known to the driver, indexed by their unique identifier ({@code host_id} in {@code
+   * system.local}/{@code system.peers}). This might include nodes that are currently viewed as
+   * down, or ignored by the load balancing policy.
    */
   @NonNull
-  Map<InetSocketAddress, Node> getNodes();
+  Map<UUID, Node> getNodes();
+
+  /**
+   * Finds the node with the given {@linkplain Node#getEndPoint() connection information}, if it
+   * exists.
+   *
+   * <p>Note that this method performs a linear search of {@link #getNodes()}.
+   */
+  @NonNull
+  default Optional<Node> findNode(@NonNull EndPoint endPoint) {
+    for (Node node : getNodes().values()) {
+      if (node.getEndPoint().equals(endPoint)) {
+        return Optional.of(node);
+      }
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Finds the node with the given <em>untranslated</em> {@linkplain Node#getBroadcastRpcAddress()
+   * broadcast RPC address}, if it exists.
+   *
+   * <p>Note that this method performs a linear search of {@link #getNodes()}.
+   */
+  @NonNull
+  default Optional<Node> findNode(@NonNull InetSocketAddress broadcastRpcAddress) {
+    for (Node node : getNodes().values()) {
+      Optional<InetSocketAddress> o = node.getBroadcastRpcAddress();
```

Comment:
```
Optional<InetSocketAddress> o = node.getBroadcastRpcAddress();
      if (o.isPresent() && o.get().equals(broadcastRpcAddress)) {
        return Optional.of(node);
      }
    }
    return Optional.empty();
```
can be simplified to one line:
`node.getBroadcastRpcAddress().filter(o -> o.equals(broadcastRpcAddress));`

Reply:
Actually the last line is outside of the loop. If `!isPresent()`, we want to keep iterating.

Reply:
Ok, so it could be re-written to:
```
  for (Node node : getNodes().values()) {
      Optional<Node> node1 = node
          .getBroadcastRpcAddress()
          .filter(o -> o.equals(broadcastRpcAddress))
          .map(t -> node);
      if(node1.isPresent()){
        return node1;
      }
    }
    return Optional.empty();
```
but doubtful it is more readable